src/Cohesity.Powershell/Scripts/Principals/Add-CohesityProtectionSourceForPrincipal.ps1

```diff
@@ -0,0 +1,119 @@
+function Add-CohesityProtectionSourceForPrincipal {
+    <#
+        .SYNOPSIS
+        Specify the security identifier (SID) of the principal to grant access permissions for protection source.
+        .DESCRIPTION
+        Add Protection Sources and Views that the specified principal has permissions to access.
+
+        .NOTES
+        Published by Cohesity
+        .LINK
+        https://cohesity.github.io/cohesity-powershell-module/#/README
+        .EXAMPLE
+        Add-CohesityProtectionSourceForPrincipal -PrincipalType "GROUP" -PrincipalName user-group1 -ProtectionSourceObjectIds 121,344
```

Comment:
Would it make sense to get the ids from an cmdlet and pipe to this or 
take by name (just like in UI) in case of multiple match/no match throw error.

Reply:
Same thing applies to all the cruds.

Reply:
Add-CohesityProtectionSourceForPrincipal, Remove-CohesityProtectionSourceForPrincipal has optional key parameters of protection source object ids and viewnames. Piping the different types of input data not feasible.

On the same line Get-CohesityUserGroup -Name user-group1, gets the user group info which matches the string(the response can have more than one entry), (the behaviour  mimicks the REST API ), therefore not feasible to pipe.

Reply:
Added piping for ***-CohesityProtectionSourceForPrincipal and ***-CohesityViewForPrincipal
Piping has no advantage with ***-CohesityUserGroup